torch/csrc/distributed/c10d/PyProcessGroup.hpp

```diff
@@ -137,4 +137,45 @@ class PyProcessGroup : public ProcessGroup {
   }
 };
 
+class TORCH_PYTHON_API PythonOnCompletionHook {
```

Comment:
This (`TORCH_API` -> `TORCH_PYTHON_API`) is the only change compared to #106988. Windows tests passed